docs/blueprints/002-extension-registry.md

```diff
@@ -0,0 +1,604 @@
+* _configuration_ elements can be present under _extension instances_, _services_, _endpoints_ or _credentials_ and represent opaque, service specific configuration data that the consumers need in order to access and consume a service interface. _Configuration_ elements can be used to encode any information relevant for service clients: accounts and credentials, information describing the service or particular parameters that describe how the service should be used. For example, if endpoints are SSL secured, custom certificates (e.g. self-signed CA certificates) might be needed to access them and this should be included in the endpoint configuration. The information encoded in a _configuration_ element is only treated as sensitive information when present under a _profile_. Equivalent of a k8s configmap (or k8s secret, when under _profile_). 
+
+Examples:
+
+1. MLFlow instance deployed locally alongside FuseML and globally accessible:
+
+  ```yaml
+  name: mlflow
+  product: mlflow
+  description: MLFlow experiment tracking service
+  instances:
+    - name: default
+      version: "1.19.0"
+      location: local
+      services:
+        - name: mlflow-tracking
+          service: mlflow-tracking
+          description: MLFlow experiment tracking service API and UI
+          endpoints:
+            - name: cluster
+              url: http://mlflow
+              type: internal
+            - name: ingress
+              url: http://mlflow.10.110.120.130.nip.io
+              type: external
+              default: True
+        - name: mlflow-store
+          service: s3
+          description: MLFlow minio S3 storage back-end
+          credentials:
+            - default-s3-account
+          endpoints:
+            - name: cluster
+              url: http://mlflow-minio:9000
+              type: internal
+            - name: ingress
+              url: http://mlflow.10.110.120.130.nip.io
+              type: external
+              default: True
+      credentials:
```

Comment:
Different instances of the same tool can have different credentials. But MLFlow is probably the best example of how you have different credentials for different services: the tracker UI can be accessed without credentials, but if you want to store models, you need credentials for the storage backend.

`scope` can be used to configure different credentials for different FuseML projects.

Reply:
Is there any reason for not moving the information under `instances/services/credentials` to `instances/services/mlflow-store/credentials` and deleting `instances/services/credentials`?

Reply:
That's a valid point, I guess we could have credentials directly under each `service` instead having them under `instance`. The question is whether that would lead to duplication or not. For example, some applications that expose multiple APIs for multiple services use a unified authentication mechanism for all those services. You can use the same user/password or token to authenticate to multiple services.

But I think those cases aren't so frequent with AI/ML related tools. If you think we can take the risk of having duplication, I agree that we can move credentials under services.

Reply:
OK, now I get it, thanks! I think this is a valid reason to keep it.